features/the_innernet/test_theinnernet.py

```diff
@@ -49,3 +70,22 @@ def test_enable_text_field(Selene: Actor) -> None:
     then(Selene).should(
         See.the(InputText.of_the(TEXT_FIELD), ContainsTheText(test_text)),
     )
+
+
+def test_disable_text_field(Selene: Actor) -> None:
+    """Test that the text field is disabled by the Disable button."""
+    given(Selene).was_able_to(
+        Open.their_browser_on(URL),
+        Wait.for_the(TEXT_FIELD).to_appear(),
+        Click.on_the(ENABLE_BUTTON),
+        Wait.for_the(WAITING_TEXT).to_appear(),
+        Wait.for_the(WAITING_TEXT).to_disappear(),
+    )
+    when(Selene).attempts_to(
+        Click.on_the(DISABLE_BUTTON),
+        Wait.for_the(WAITING_TEXT).to_appear(),
+        Wait.for_the(WAITING_TEXT).to_disappear(),
+    )
+    then(Selene).should(
+        See.the(Element(TEXT_FIELD), IsNot(Enabled()))
```

Comment:
I wrote `IsEnabled` as a new Resolution (Screenpy's terminology for an assertion matching function). It matches against elements where selenium's `isEnabled = True`. Full disclosure - after I went to the trouble, I realized one of the existing resolutions in the base framework, `IsClickable`, actually matches when an element is both `.isVisible` and `.isEnabled`, which would have worked just fine here, but I decided to keep my custom one just for demonstration purposes.